### ISSUE TITLE
goog.module: goog.require without assignment causes spurious warnings.

### ISSUE THREAD

**Issue by camio** [google/closure-compiler#1203]
For example the code here, http://pastie.org/10469001, produces the following warnings

```
cwf/base/CheckedInputElement_test.js|4| WARNING - Suspicious code. The result of the 'getprop' operator is not being used.
|| goog.require( 'cwf.base.CheckedInputElement' );
|| ^
|| 
cwf/base/CheckedInputElement_test.js|5| WARNING - Suspicious code. The result of the 'getprop' operator is not being used.
|| goog.require( 'goog.testing.asserts' );
|| ^
|| 
cwf/base/CheckedInputElement_test.js|6| WARNING - Suspicious code. The result of the 'getprop' operator is not being used.
|| goog.require( 'goog.testing.events' );
|| ^
|| 
cwf/base/CheckedInputElement_test.js|7| WARNING - Suspicious code. The result of the 'getprop' operator is not being used.
|| goog.require( 'goog.testing.jsunit' );
|| ^
|| 
```

It seems like this is due to 'goog.require' commands being transformed to 'getprop' commands.  'getprop' is considered to be a pure operator which has no side-effects; thus, the warning.

**Comment by camio** [comment]
The actual code that does the replacement is in 'src/com/google/javascript/jscomp/ClosureRewriteModule.java' at line 470:

```
call.getParent().replaceChild(call, replacement);
```

If I replace this with a line that can possibly have a side-effect, the spurious warning goes away:

```
call.getParent().replaceChild(call, IR.or( replacement, IR.call( IR.name("eval"), IR.string("null"))));
```

I'm not sure how this can be done cleanly though.

**Comment by MatrixFrog** [comment]
Fixed in 4398f4e047971c192178cb07d68da822f2871306
